src/components/Input.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { View, StyleSheet, TextInput } from 'react-native';
+
+const styles = StyleSheet.create({
+  input: {
+    borderTopWidth: 0,
+    borderRightWidth: 0,
+    borderLeftWidth: 0,
+    borderBottomWidth: 2,
+    borderColor: '#ccc',
+    paddingTop: 15,
+    paddingBottom: 15,
+    marginLeft: 15,
+    marginRight: 15,
+    color: '#000'
+  },
+  withIcon: {
+    paddingLeft: 35
+  }
+});
+
+export type InputProps = {
+  value: string
+  onChange: any
+  placeholder: string,
+  editable?: boolean,
+  children?: any
+}
+
+export default function Input(props: InputProps) {
+  const { value, onChange, placeholder, editable, children: Icon } = props;
```

Comment:
`children` is something from internal React concepts, you don't usually want to rename it. Is it mandatory?